staging/src/k8s.io/kube-aggregator/pkg/apiserver/apiserver.go

```diff
@@ -249,31 +219,29 @@ func (c completedConfig) NewWithDelegate(delegationTarget genericapiserver.Deleg
 		return nil
 	})
 
-	s.GenericAPIServer.PrepareOpenAPIService()
-
-	if s.GenericAPIServer.OpenAPIService != nil {
-		s.rootSpec = s.GenericAPIServer.OpenAPIService.GetSpec()
-		if err := s.updateOpenAPISpec(); err != nil {
+	if openApiConfig != nil {
+		s.openAPIAggregator, err = buildAndRegisterOpenAPIAggregator(
+			s.delegateHandler,
```

Comment:
I added an issue for that. I think we should CRDs need to register themselves with aggregator api server. Whatever we decide, though, I think it should be in another PR. The purpose of this PR was reintegrating the same functionality of the original PR. Do you agree?